tests/test_line_box.py

```diff
@@ -0,0 +1,187 @@
+            [
+                "╭─────╮",
+                "│ Text│",
+                "│  on │",
+                "│ four│",
+                "│lines│",
+                "╰─────╯",
+            ],
+            [line.decode("utf-8") for line in canvas.text],
+        )
+
+    def test_header(self):
+        wrapped = urwid.Text("Some text")
+        l = urwid.LineBox(wrapped, title="Title", tlcorner="╭", trcorner="╮", blcorner="╰", brcorner="╯")
+        cols, rows = 11, 3
+        self.assertEqual((cols, rows), l.pack(()))
+
+        canvas = l.render(())
+        self.assertEqual(cols, canvas.cols())
+        self.assertEqual(rows, canvas.rows())
+
+        self.assertEqual(
+            [
+                "╭─ Title ─╮",
+                "│Some text│",
+                "╰─────────╯",
+            ],
+            [line.decode("utf-8") for line in canvas.text],
+        )
+
+    def test_negative(self):
+        wrapped = urwid.Text("")
+        with self.assertRaises(ValueError) as ctx:
+            l = urwid.LineBox(wrapped, title="Title", tline="")
+
+        self.assertEqual("Cannot have a title when tline is empty string", str(ctx.exception))
+
+    def test_partial_contour(self):
+        def mark_pressed(btn: urwid.Button) -> None:
+            nonlocal pressed
+
+            pressed = True
+
+        pressed = False
+        wrapped = urwid.Button("Wrapped", on_press=mark_pressed)
+
+        with self.subTest("No top line -> no top"):
+            l = urwid.LineBox(wrapped, tline="")
+            cols, rows = 13, 2
+            self.assertEqual((cols, rows), l.pack(()))
+
+            canvas = l.render(())
+            self.assertEqual(cols, canvas.cols())
+            self.assertEqual(rows, canvas.rows())
+
+            self.assertEqual(
+                [
+                    "│< Wrapped >│",
+                    "└───────────┘",
+                ],
+                [line.decode("utf-8") for line in canvas.text],
+            )
+            self.assertIsNone(l.keypress((), "enter"))
+            self.assertTrue(pressed)
+
+        pressed = False
+
+        with self.subTest("No right side elements -> no side"):
+            l = urwid.LineBox(wrapped, trcorner="", rline="", brcorner="")
+            cols, rows = 12, 3
+            self.assertEqual((cols, rows), l.pack(()))
+
+            canvas = l.render(())
+            self.assertEqual(cols, canvas.cols())
+            self.assertEqual(rows, canvas.rows())
+
+            self.assertEqual(
+                [
+                    "┌───────────",
+                    "│< Wrapped >",
+                    "└───────────",
+                ],
+                [line.decode("utf-8") for line in canvas.text],
+            )
+            self.assertIsNone(l.keypress((), "enter"))
+            self.assertTrue(pressed)
+
+        pressed = False
+
+        with self.subTest("No left side elements -> no side"):
+            l = urwid.LineBox(wrapped, tlcorner="", lline="", blcorner="")
```

Comment:
do we need to remove corners when lline is empty?

Reply:
I mean it would be a nicer interface if we didn't, like with top and bottom

Reply:
We can add it to make consistent with top and bottom

Reply:
Done